x/concentrated-liquidity/position.go

```diff
@@ -401,3 +436,49 @@ func (k Keeper) validatePositionsAndGetTotalLiquidity(ctx sdk.Context, owner sdk
 	}
 	return basePosition.PoolId, basePosition.LowerTick, basePosition.UpperTick, totalLiquidity, nil
 }
+
+// GetConcentratedLockupDenom returns the concentrated lockup denom for a given pool and position.
+func GetConcentratedLockupDenom(poolId, positionId uint64) string {
+	return fmt.Sprintf("cl/pool/%d/%d", poolId, positionId)
+}
+
+// GetPositionIdToLock returns the positionId to lock mapping in state.
+func (k Keeper) GetPositionIdToLock(ctx sdk.Context, positionId uint64) (uint64, error) {
+	store := ctx.KVStore(k.storeKey)
+
+	// Get the position ID to key mapping.
+	key := types.KeyPositionIdForLock(positionId)
+	value := store.Get(key)
+	if value == nil {
+		return 0, types.PositionIdToLockNotFoundError{PositionId: positionId}
+	}
+
+	return sdk.BigEndianToUint64(value), nil
+}
+
+// SetPositionIdToLock sets the positionId to lock mapping in state.
+func (k Keeper) SetPositionIdToLock(ctx sdk.Context, positionId, underlyingLockId uint64) {
+	store := ctx.KVStore(k.storeKey)
+
+	// Get the position ID to key mapping.
+	key := types.KeyPositionIdForLock(positionId)
+	store.Set(key, sdk.Uint64ToBigEndian(underlyingLockId))
+}
```

Comment:
Is this new index added to genesis? If not, can you please do so in this PR

Reply:
Discussed adding in separate PR but please add it to the issue and a TODO with link to issue in code

Reply:
The issue was already created, but added the TODO in code here https://github.com/osmosis-labs/osmosis/pull/4882/commits/8436cd092f70047d37b9133211a06a087320ac28